pageserver/src/remote_storage/storage_sync.rs

```diff
@@ -365,19 +368,41 @@ async fn fetch_existing_uploads<P: std::fmt::Debug, S: 'static + RemoteStorage<S
     let mut data_fetches = uploaded_files
         .into_iter()
         .map(|remote_path| async {
-            (
-                remote_file_info(remote_storage, &remote_path).await,
-                remote_path,
-            )
+            let local_path = match remote_storage.local_path(&remote_path) {
```

Comment:
It'd be cool to reduce the duplication of map_err in this function. Maybe use a helper function and `?`.

Reply:
Agree, but let's omit that in this PR: the code for fetching remote files is different with archives, so does not make too much sense to make this version readable.